ddtrace/internal/serverless/__init__.py

```diff
@@ -28,3 +29,19 @@ def in_gcp_function():
     is_deprecated_gcp_function = environ.get("FUNCTION_NAME", "") != "" and environ.get("GCP_PROJECT", "") != ""
     is_newer_gcp_function = environ.get("K_SERVICE", "") != "" and environ.get("FUNCTION_TARGET", "") != ""
     return is_deprecated_gcp_function or is_newer_gcp_function
+
+
+def in_azure_function_consumption_plan():
+    # type: () -> bool
+    """Returns whether the environment is an Azure Consumption Plan Function.
+    This is accomplished by checking the presence of two Azure Function env vars,
+    as well as a third SKU variable indicating consumption plans.
+    """
+    is_azure_function = (
+        os.environ.get("FUNCTIONS_WORKER_RUNTIME", "") != "" and os.environ.get("FUNCTIONS_EXTENSION_VERSION", "") != ""
+    )
+
+    website_sku = os.environ.get("WEBSITE_SKU", "")
+    is_consumption_plan = website_sku == "" or website_sku == "Dynamic"
+
+    return is_azure_function and is_consumption_plan
```

Comment:
Would it be useful to split up these checks? Do you envision a scenario where we would like to just call `is_azure_function()`?

Reply:
I think we're good for now. `is_azure_function and !is_consumption_plan` is useful for identifying azure functions running on app service (non consumption) plans. We won't need this as we have a separate non mini-agent way of supporting python on app service plans!